integration/consul_catalog_test.go

```diff
@@ -98,16 +115,11 @@ func (s *ConsulCatalogSuite) TestSingleService(c *check.C) {
 	c.Assert(err, checker.IsNil, check.Commentf("Error registering service"))
 	defer s.deregisterService("test", nginx.NetworkSettings.IPAddress)
 
-	time.Sleep(5000 * time.Millisecond)
-	client := &http.Client{}
 	req, err := http.NewRequest("GET", "http://127.0.0.1:8000/", nil)
 	c.Assert(err, checker.IsNil)
 	req.Host = "test.consul.localhost"
-	resp, err := client.Do(req)
-
-	c.Assert(err, checker.IsNil)
-	c.Assert(resp.StatusCode, checker.Equals, 200)
 
-	_, err = ioutil.ReadAll(resp.Body)
+	cond := utils.ComposeCondition(utils.UntilStatusCodeIs(200), utils.BodyContains("Welcome to nginx!"))
+	err = utils.TryRequest(req, 5*time.Second, cond)
```

Comment:
I would have make `TryRequest` taking *variadic args*, so that you could do `TryRequest(req, 5*time.Second, utils.UntilStatusCodeIs…, utils.Body…)`.